File: leetcode2/1easy/최원준/Q3270.java
package Leetcode;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

public class Q3270 {
    class Solution {
        public String getString(int num) {
            String n1 = String.valueOf(num);
            while (n1.length() < 4) {
                n1 = "0" + n1;
            }
            return n1;
        }
        public int generateKey(int num1, int num2, int num3) {
            String n1 = getString(num1);
            String n2 = getString(num2);
            String n3 = getString(num3);

            StringBuilder sb = new StringBuilder();
            for (int i=0; i<4; i++) {
                sb.append( Math.min(
                        Integer.parseInt(n1.charAt(i)+""), Math.min(
                                Integer.parseInt(n2.charAt(i)+""),
                                Integer.parseInt(n3.charAt(i)+""))
                ));
            }

            return Integer.parseInt(sb.toString());
        }
    }
}

File: leetcode2/1easy/강병호/Q3270.py
class Leetcode.이재훈.Solution:
    def generateKey(self, num1: int, num2: int, num3: int) -> int:
        num1 = str(num1)
        num2 = str(num2)
        num3 = str(num3)

        num1 = num1.rjust(4, '0')
        num2 = num2.rjust(4, '0')
        num3 = num3.rjust(4, '0')

        result = ""

        for i in range(0, 4):
            minimum = min(int(num1[i]), int(num2[i]), int(num3[i]))

            result += str(minimum)
        

        return int(result)

if __name__ == "__main__":
    sol = Leetcode.이재훈.Solution()
    print(sol.generateKey(1, 10, 1000))
    print(sol.generateKey(1, 2, 3))
    print(sol.generateKey(1987, 879, 798))